presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2062,17 +2062,22 @@ else if (partitionUpdate.getUpdateMode() == NEW || partitionUpdate.getUpdateMode
                 // Track the manifest blob size
                 getManifestSizeInBytes(session, partitionUpdate, extraPartitionMetadata).ifPresent(hivePartitionStats::addManifestSizeInBytes);
 
+                boolean existingPartition = existingPartitions.containsKey(partitionUpdate.getName());
+                Optional<Long> existingPartitionVersion = Optional.empty();
+                if (existingPartition && partitionUpdate.getUpdateMode() == OVERWRITE) {
+                    existingPartitionVersion = existingPartitions.get(partitionUpdate.getName()).map(Partition::getPartitionVersion).orElse(Optional.empty());
```

Comment:
`Partition::getPartitionVersion` return `Optional<Long> `, we dont need the `.orElse(Optional.empty())` here

Reply:
orElse is to return empty if partition does not exist  , if we remove orElse, the return type will be Optional<Optional<Long>>.

Reply:
we can use flatMap . like this
`existingPartitionVersion = existingPartitions.get(partitionUpdate.getName()).flatMap(Partition::getPartitionVersion);
`